x-pack/plugins/security/server/authentication/providers/saml.ts

```diff
@@ -354,46 +334,24 @@ export class SAMLAuthenticationProvider extends BaseAuthenticationProvider {
         : 'Login has been initiated by Identity Provider.'
```

Comment:
> Are there any nuances between IdP and SP initiated logins that we should be aware of?

Cannot think of anything that we should worry about. The only case that seems confusing to the user is when IdP initiated login is performed with a valid intermediate cookie from not finished SP initiated login - in this case we fail and it sounds correct since we don't parse SAML responses in Kibana to be able to figure out the intention.

> [2020-07-30T19:03:10.493Z][DEBUG][plugins.security.authenticator] Session is authenticated, existing unauthenticated session will be invalidated.
> [2020-07-30T19:03:11.265Z][DEBUG][plugins.security.session] Successfully invalidated existing session.

These look redundant though, not dangerous, but confusing since session doesn't exist at this stage - I'll fix that, thanks!